models/rarible/polygon/rarible_polygon_events.sql

```diff
@@ -0,0 +1,178 @@
+            ELSE '0x' || right(substring(leftAsset:data, 3, 64), 40)
+        END AS currency_contract,
+        newRightFill AS amount_raw
+    FROM {{ source ('rarible_polygon', 'Exchange_evt_Match') }}
+    WHERE leftAsset:assetClass in ('0xaaaebeba', '0x8ae85d84') -- 0xaaaebeba: MATIC; 0x8ae85d84: ERC20 TOKEN
+        {% if not is_incremental() %}
+        AND evt_block_time >= '{{nft_start_date}}'
+        {% endif %}
+        {% if is_incremental() %}
+        AND evt_block_time >= date_trunc("day", now() - interval '1 week')
+        {% endif %}
+),
+
+contract_list as (
+    SELECT distinct nft_contract_address
+    FROM trades
+),
+
+mints AS (
+    SELECT 'mint' AS trade_category,
+        block_time AS evt_block_time,
+        block_number AS evt_block_number,
+        tx_hash AS evt_tx_hash,
+        CAST(NULL AS string) AS contract_address, -- We leave it NULL here and will get its value below by join from transactions table
+        evt_index,
+        'Mint' AS evt_type,
+        `to` AS buyer,
+        CAST(NULL AS string) AS seller,
+        contract_address AS nft_contract_address,
+        token_id,
+        amount AS number_of_items,
+        token_standard,
+        '0x385eeac5cb85a38a9a07a70c73e0a3271cfb54a7' AS currency_contract, -- All sale are in GHST
+        CAST(0 as DECIMAL(38,0)) AS amount_raw -- It's hard to get the mint price. So handle it similar as in nftb_bnb_events
+    FROM {{ ref('nft_polygon_transfers') }}
+    WHERE contract_address IN ( SELECT nft_contract_address FROM contract_list )
+        AND `from` = '0x0000000000000000000000000000000000000000'   -- mint
+        {% if not is_incremental() %}
+        AND block_time >= '{{nft_start_date}}'
+        {% endif %}
+        {% if is_incremental() %}
+        AND block_time >= date_trunc("day", now() - interval '1 week')
+        {% endif %}
+),
+
+all_events as (
+    SELECT * FROM mints
+    UNION ALL
+    SELECT * FROM trades
+),
+
+price_list AS (
+    SELECT contract_address,
+          minute,
+          price,
+          decimals,
+          symbol
+     FROM {{ source('prices', 'usd') }} p
+     WHERE blockchain = 'polygon'
+        AND contract_address IN ( SELECT DISTINCT currency_contract FROM trades) 
+        {% if not is_incremental() %}
+        AND minute >= '{{nft_start_date}}' 
+        {% endif %}
+        {% if is_incremental() %}
+        AND minute >= date_trunc("day", now() - interval '1 week')
+        {% endif %}
+) 
+
+SELECT
+  'polygon' AS blockchain,
+  'rarible' AS project,
+  'v1' AS version,
+  a.evt_tx_hash AS tx_hash,
+  date_trunc('day', a.evt_block_time) AS block_date,
+  a.evt_block_time AS block_time,
+  a.evt_block_number AS block_number,
+  amount_raw / power(10, p.decimals) * p.price AS amount_usd,
+  amount_raw / power(10, p.decimals) AS amount_original,
+  amount_raw,
+  CASE WHEN p.symbol = 'WMATIC' THEN 'MATIC' ELSE p.symbol END AS currency_symbol,
```

Comment:
For decimals and symbol you should use the tokens.erc20 table instead of the prices table as a source.

Not all tokens might have prices, but we'd still want the symbol and the correct decimals in that case.

Reply:
Updated.